src/java/org/apache/cassandra/utils/FailingBiFunction.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.utils;
+
+import java.util.function.BiFunction;
+
+public interface FailingBiFunction<T, U, R> extends BiFunction<T, U, R>
+{
+    R applyOrFail(T t, U u) throws Throwable;
+
+    @Override
+    default R apply(T t, U u)
+    {
+        try
+        {
+            return applyOrFail(t, u);
+        }
+        catch (Throwable e)
+        {
+            throw Throwables.throwAsUncheckedException(e);
+        }
+    }
+
+    static <T, U, R> FailingBiFunction<T, U, R> orFail(FailingBiFunction<T, U, R> fn)
```

Comment:
nit: Anything use `orFail()`?

Reply:
in this patch no, I normally add it so lambdas don't need to cast...  its dead code; I can always remove... the other `Failing*` classes have this